mapeditor/mapsettings.cpp

```diff
@@ -108,5 +454,423 @@ void MapSettings::on_pushButton_clicked()
+				specialVictory.trigger = EventExpression(cond);
+				break;
+			}
+				
+		}
+		
+		// if condition is human-only turn it into following construction: AllOf(human, condition)
+		if(ui->onlyForHumansCheck->isChecked())
+		{
+			EventExpression::OperatorAll oper;
+			EventCondition notAI(EventCondition::IS_HUMAN);
+			notAI.value = 1;
+			oper.expressions.push_back(notAI);
+			oper.expressions.push_back(specialVictory.trigger.get());
+			specialVictory.trigger = EventExpression(oper);
+		}
+
+		// if normal victory allowed - add one more quest
+		if(ui->standardVictoryCheck->isChecked())
+		{
+			controller.map()->victoryMessage += " / ";
+			controller.map()->victoryMessage += VLC->generaltexth->victoryConditions[0];
+			controller.map()->triggeredEvents.push_back(standardVictory);
+		}
+		controller.map()->triggeredEvents.push_back(specialVictory);
+	}
+	
+	//DEFEAT
+	if(ui->loseComboBox->currentIndex() == 0)
+	{
+		controller.map()->triggeredEvents.push_back(standardDefeat);
+		controller.map()->defeatIconIndex = 3;
+		controller.map()->defeatMessage = VLC->generaltexth->lossCondtions[0];
+	}
+	else
+	{
+		int lossCondition = ui->loseComboBox->currentIndex() - 1;
+		
+		TriggeredEvent specialDefeat;
+		specialDefeat.effect.type = EventEffect::DEFEAT;
+		specialDefeat.identifier = "specialDefeat";
+		specialDefeat.description.clear(); // TODO: display in quest window
+		
+		controller.map()->defeatIconIndex = lossCondition;
+		controller.map()->defeatMessage = VLC->generaltexth->lossCondtions[size_t(lossCondition) + 1];
+		
+		switch(lossCondition)
+		{
+			case 0: {
+				EventExpression::OperatorNone noneOf;
+				EventCondition cond(EventCondition::CONTROL);
+				cond.objectType = Obj::TOWN;
+				assert(loseTypeWidget);
+				int townIdx = loseTypeWidget->currentData().toInt();
+				cond.position = controller.map()->objects[townIdx]->pos;
+				noneOf.expressions.push_back(cond);
+				specialDefeat.onFulfill = VLC->generaltexth->allTexts[251];
+				specialDefeat.trigger = EventExpression(noneOf);
+				break;
+			}
+				
+			case 1: {
+				EventExpression::OperatorNone noneOf;
+				EventCondition cond(EventCondition::CONTROL);
+				cond.objectType = Obj::HERO;
+				assert(loseTypeWidget);
+				int townIdx = loseTypeWidget->currentData().toInt();
+				cond.position = controller.map()->objects[townIdx]->pos;
+				noneOf.expressions.push_back(cond);
+				specialDefeat.onFulfill = VLC->generaltexth->allTexts[253];
+				specialDefeat.trigger = EventExpression(noneOf);
+				break;
+			}
+				
+			case 2: {
+				EventCondition cond(EventCondition::DAYS_PASSED);
+				assert(loseValueWidget);
+				cond.value = expiredDate(loseValueWidget->text());
+				specialDefeat.onFulfill = VLC->generaltexth->allTexts[254];
+				specialDefeat.trigger = EventExpression(cond);
+				break;
+			}
+				
+			case 3: {
+				EventCondition cond(EventCondition::DAYS_WITHOUT_TOWN);
+				assert(loseValueWidget);
+				cond.value = loseValueWidget->text().toInt();
```

Comment:
I assume that vcmi supports custom values for "DAYS_WITHOUT_TOWN"?

Reply:
Yes, this one is customizable, so you can try to make a map with heroes only without towns